patroni/__init__.py

```diff
@@ -48,13 +48,19 @@ def load_dynamic_configuration(self):
                 logger.warning('Can not get cluster from dcs')
 
     def get_tags(self):
-        return {tag: value for tag, value in self.config.get('tags', {}).items()
+        tags = {tag: value for tag, value in self.config.get('tags', {}).items()
                 if tag not in ('clonefrom', 'nofailover', 'noloadbalance') or value}
+        tags.update(self.ha.get_dynamic_tags())
```

Comment:
Hmm, not sure how to best handle this.

1. Update tags after all callsites that could affect dynamic tags (bug prone)
2. Make patroni.tags a property (lots of useless dict building to check if some flag is set or not)
3. Make `Ha.touch_member` apply dynamic tags. (creates a separation about publicly visible and local tags that might get confusing in the future)
4. Have a separate member attribute for dynamic nosync override and move the `_disable_sync` check into `touch_member`, have `pick_synchronous_standby` check that attribute. (simple and stupid, non-general, makes DCS API wider for a single redundant bit of information)

I'm currently thinking 3, but any other ideas are welcome.